src/combat.ts

```diff
@@ -615,46 +619,90 @@ export class Macro extends StrictMacro {
+          get("lastCopyableMonster") === $monster`Knob Goblin Embezzler` &&
+          myFamiliar() === meatFamiliar(),
+        Macro.skill($skill`Back-Up to your Last Enemy`).step(Macro.load()),
+        Macro.basicCombat()
+      )
+    ).step(macro);
+  } else {
+    return macro;
+  }
+}
+
+function makeCcs<M extends StrictMacro>(macro: M) {
+  writeCcs(`[default]\n"${customizeMacro(macro).toString()}"`, "garbo");
+  cliExecute("ccs garbo");
+}
+
+function completeCombat<T>(action: () => T) {
+  try {
+    const result = action();
+    while (inMultiFight()) runCombat();
+    if (choiceFollowsFight()) visitUrl("choice.php");
+    return result;
+  } catch (e) {
+    throw `Combat exception! Last macro error: ${get("lastMacroError")}. Exception ${e}.`;
+  }
+}
+
 /**
  * Attempt to perform a nonstandard combat-starting Action with a Macro
  * @param macro The Macro to attempt to use
  * @param action The combat-starting action to attempt
  * @param tryAuto Whether or not we should try to resolve the combat with an autoattack; autoattack macros can fail against special monsters, and thus we have to submit a macro with Macro.save() regardless
  * @returns The output of your specified action function (typically void)
  */
-export function withMacro<T>(macro: Macro, action: () => T, tryAuto = false): T {
+export function withMacro<T, M extends StrictMacro>(macro: M, action: () => T, tryAuto = false): T {
   if (getAutoAttack() !== 0) setAutoAttack(0);
   if (tryAuto) macro.setAutoAttack();
-  macro.save();
+  makeCcs(macro);
   try {
     return action();
   } finally {
-    Macro.clearSaved();
     if (tryAuto) setAutoAttack(0);
   }
 }
 
-export function main(): void {
-  if (have($effect`Eldritch Attunement`)) {
-    Macro.if_($monster`Eldritch Tentacle`, Macro.basicCombat())
-      .step(Macro.load())
-      .submit();
-  } else if (getTodaysHolidayWanderers().length !== 0) {
-    Macro.ifHolidayWanderer(
-      Macro.externalIf(
-        haveEquipped($item`backup camera`) &&
-          get("_backUpUses") < 11 &&
-          get("lastCopyableMonster") === $monster`Knob Goblin Embezzler` &&
-          myFamiliar() === meatFamiliar(),
-        Macro.skill($skill`Back-Up to your Last Enemy`).step(Macro.load()),
-        Macro.basicCombat()
-      )
-    )
-      .step(Macro.load())
-      .submit();
-  } else {
-    Macro.load().submit();
-  }
-  while (inMultiFight()) runCombat();
-  if (choiceFollowsFight()) visitUrl("choice.php");
+/**
+ * Adventure in a location and handle all combats with a given macro.
+ * To use this function you will need to create a consult script that runs Macro.load().submit() and a CCS that calls that consult script.
+ * See examples/consult.ts for an example.
+ *
+ * @category Combat
+ * @param loc Location to adventure in.
+ * @param macro Macro to execute.
+ */
+export function garboAdventure<M extends StrictMacro>(loc: Location, macro: M): void {
+  if (getAutoAttack() !== 0) setAutoAttack(0);
```

Comment:
We might want to always call this; I've seen some issues with the `aa` cli command that veracity detailed [here](https://kolmafia.us/threads/mafia-appears-not-to-be-tracking-_spacejellyfishdrops-when-i-autoattack-with-a-macro-r26881.28212/#post-170343).

Reply:
I think that triggers an extra server request?